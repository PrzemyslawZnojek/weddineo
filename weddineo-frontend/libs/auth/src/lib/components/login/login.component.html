<div [fxFlex]="'100%'" [fxLayoutAlign]="'center center'" class="container">
  <mat-card [fxFlex.gt-xs]="'500px'" [fxFlex]="'80%'">
    <mat-card-content [fxLayout]="'column'">
      <weddineo-frontend-text-input
        [label]="'auth.login.login' | translate"
        [fc]="formGroup.get('login')"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <weddineo-frontend-text-input
        [label]="'auth.login.password' | translate"
        [fc]="formGroup.get('password')"
        [type]="'password'"
        [errors]="errors"
      >
      </weddineo-frontend-text-input>
      <weddineo-frontend-action-button
        [label]="'auth.login.signIn' | translate"
        [width]="'150px'"
        [fxLayoutAlign]="'center center'"
        (onClick)="login()"
      >
      </weddineo-frontend-action-button>
    </mat-card-content>
  </mat-card>
</div>
