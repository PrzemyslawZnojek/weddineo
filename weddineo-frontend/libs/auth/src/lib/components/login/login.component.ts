import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { ErrorMessage } from 'libs/ui-kit/src/lib/model/error-message';
import { AuthFacade } from '../../+state/auth.facade';

@Component({
  selector: 'weddineo-frontend-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.scss'],
})
export class LoginComponent implements OnInit {
  formGroup: FormGroup;

  constructor(private fb: FormBuilder, private authFacade: AuthFacade) {}

  ngOnInit(): void {
    this.initForm();
  }

  private initForm() {
    this.formGroup = this.fb.group({
      login: [null, Validators.required],
      password: [null, Validators.required]
    });
  }

  errors: ErrorMessage[] = [
    {
      error: 'required',
      message: 'Field is required'

    }
  ]

  login(){
    this.authFacade.login({
      login: this.formGroup.get('login').value,
      password: this.formGroup.get('password').value
    })
  }
}
