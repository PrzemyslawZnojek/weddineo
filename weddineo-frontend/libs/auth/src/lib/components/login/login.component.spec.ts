import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MockComponents } from 'ng-mocks';
import { LoginComponent } from './login.component';
import {
  ActionButtonComponent,
  TextInputComponent,
} from '@weddineo-frontend/ui-kit';
import { MatCardModule } from '@angular/material/card';
import { FlexLayoutModule } from '@angular/flex-layout';
import { ReactiveFormsModule } from '@angular/forms';


describe('LoginComponent', () => {
  let component: LoginComponent;
  let fixture: ComponentFixture<LoginComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MatCardModule, FlexLayoutModule, ReactiveFormsModule],
      declarations: [
        LoginComponent,
        MockComponents(ActionButtonComponent, TextInputComponent),
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render elements', () => {
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.querySelectorAll('weddineo-frontend-text-input').length).toEqual(2);
    expect(compiled.querySelectorAll('weddineo-frontend-action-button').length).toEqual(
      1
    );
  });
});
