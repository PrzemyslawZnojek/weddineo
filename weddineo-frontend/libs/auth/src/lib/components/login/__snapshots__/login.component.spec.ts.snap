// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match snapshot 1`] = `
<weddineo-frontend-login
  errors={[Function Array]}
  fb={[Function FormBuilder]}
  formGroup={[Function FormGroup]}
>
  <div
    class="container"
    style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: center; flex: 1 1 100%; max-width: 100%;"
  >
    <mat-card
      class="mat-card mat-focus-indicator"
      style="box-sizing: border-box; max-width: 80%; flex: 1 1 100%;"
    >
      <mat-card-content
        class="mat-card-content"
        style="box-sizing: border-box; display: flex; flex-direction: column;"
      >
        <weddineo-frontend-text-input />
        <weddineo-frontend-text-input />
        <weddineo-frontend-action-button
          style="align-content: center; align-items: center; box-sizing: border-box; display: flex; flex-direction: row; justify-content: center;"
        />
      </mat-card-content>
    </mat-card>
  </div>
</weddineo-frontend-login>
`;
